### PR TITLE
MINOR: [Docs] Add link to ListView format in pyarrow docs

### PR DESCRIPTION
**Pull request by danepitkin**
### Rationale for this change

We should link to the list view format when referencing this new layout in pyarrow usage docs.

### What changes are included in this PR?

* Add a link to the listview layout

### Are these changes tested?

Yes, I generated docs and verified the link.

### Are there any user-facing changes?

No.

### CODE REVIEW

**Comment by danepitkin** [comment]
Manually verified:

image
image

File: docs/source/python/data.rst [hunk 1]
```diff
@@ -252,6 +252,8 @@ This allows for ListView arrays to specify out-of-order offsets:
    arr = pa.ListViewArray.from_arrays(offsets, sizes, values)
    arr
 
+See the format specification for more details on the :ref:`listview-layout`.
```
Comment (felipecrv) [inline docs/source/python/data.rst:255]: is the `the` redundant?

**Review by felipecrv — APPROVED** [review]

File: docs/source/python/data.rst (hunk 1 above)
Comment (danepitkin) [reply docs/source/python/data.rst:255]: +1 I'll remove it!

**Review by AlenkaF — APPROVED** [review]
Thanks for fixing header underlining!

**Pull request merged**